Model/Api/UpdateCart.php

```diff
@@ -223,6 +223,17 @@ public function execute($cart, $add_items = null, $remove_items = null, $discoun
 
             $result = $this->generateResult($immutableQuote);
 
+            //this is only true for universal api requests, in which case we want to return the result and
+            //let UniversalApi::execute deal with sending the response
+            if (isset($this->request->getBodyParams()['data'])) {
```

Comment:
Should we have early return in this IF?

Reply:
I had to remove it and start sending the response from here even for the universal endpoint due to issues with the DataObjectProcessor and how it would parse the objects in the returned UniversalApiInterface. This is something that will be addressed in MA-493, but here I simply forgot to update the comment.